vcf_slice: Fix basic checklist items for DX apps

msimbirsky opened the pull request:
#dxapp.json
- [x] Upgrade to 14.04
- [x] Add help and extension to input spec.
- [x] In `details` field specify: `whatsNew`and `citation` (leave blank for citation.
- [x] In `details` consider specifying `upstreamAuthor` and `upstreamVersion` fields when applicable.
- [x] Use `regionalOptions` 
- [x] Set applicable timeout (Be relax) and restart policy in [`runSpec`](https://wiki.dnanexus.com/API-Specification-v1.0.0/IO-and-Run-Specifications).
- [x] Add subjob restartability when applicable.
- [x] `version` field, set proper [semantic versioning](https://semver.org/).
- [x] Setting `"openSource": true`. 
- [x] Run [dxapp linter](https://github.com/dnanexus/RoughER/blob/master/scripts/dxapp_lint.py) [[permalink](https://github.com/dnanexus/RoughER/blob/ecb3ea3f97870c020f93bf4723afc9f40a310e33/scripts/dxapp_lint.py)] on `dxapp.json`.
- [x] Use AssetDepend instead of execDepend
- [x] Add default instance type that works for a general use case.
- [x] add new input if applicable

#src/code.sh src/code.py
- [x] GATK tarball should be removed from the resource project (or make a new resource project), and the applet should require GATK tarball as a required input. This way, the users are enforced to provide their own tarball. This would avoid trouble from non-academic entities without GATK license to use your tarball.

# Readme.md
- [x] Improve Readme file to contain applet function, use case, input/output description, underline execution if it’s a wrapper.
- [x] Read documentation for sanity check. (Unpublished app URL will be: https://platform.dnanexus.com/app/app-xxxx)

# Test
- [x] Basic testing using given input in Box or expected input
- [ ] Make a Pull Request to `master` in https://github.com/RitchieLab/dnanexus

# Production
- [ ] Co-test with Ritchie group
- [ ] Help Ritchie group to publish (Give dx command)
- [ ] Verify the versions that appear on the platform are the intended versions
- [ ] Set up demo workflow in public project.

etal committed:
vcf_slice: Fix basic checklist items for DX apps

- dxapp.json: Add input help text, citation stub, regions, timeout, distro
- Note that GNU parallel should be in an asset bundle (TODO)
- Consistent whitespace/indentation in the main script
- Remove dead code
- Remove boilerplate comments from Readme.md and main script
- Set shell options '-e -o pipefail'

msimbirsky commented:
Notes (from Eric?)
- dxapp.json: Add input help text, citation stub, regions, timeout, distro
- Note that GNU parallel should be in an asset bundle (TODO)
- Consistent whitespace/indentation in the main script
- Remove dead code
- Remove boilerplate comments from Readme.md and main script
- Set shell options '-e -o pipefail'

Commit:
Remove authorizedUsers and developers from dxapp.json

Commit:
Update asset depends to use shared asset

Commit:
Update 'suffix_fn' input to 'suffix'

Since this input is a string and not a file, this name should be less
confusing

Commit:
Specify file type for regions file

Commit:
Remove duplicate "details field"

Commit:
Remove extraneous dx describe commands

Since all the info we need is available locally, this will make it
easier on our API if a lot of files are run.
This seems to break the app due to a newline character somewhere --
still need to troubleshoot.

Commit:
Update vcf_slice app

- Add pattern to outputs
- Update readme (TODO: more detail)
- Fix newline error introduced in last commit
- Add check for regions file

Commit:
Update help strings to be linter compatible

Commit:
Update details to include GPL license

Commit:
Update readme file

msimbirsky commented:
Pull request ready for review

yihchii commented on sequencing/vcf_slice/dxapp.json:
```diff
@@ -1,69 +1,90 @@
+  "summary": "Return a small section of a VCF file or files (similar to tabix).",
   "dxapi": "1.0.0",
-  "version": "0.0.1",
+  "version": "0.1.0",
```
"version": "0.1.0", -> "version": "1.0.0",

yihchii commented on sequencing/vcf_slice/dxapp.json:
```diff
@@ -1,69 +1,90 @@
   ],
   "inputSpec": [
     {
       "name": "vcf_fn",
```
"vcf_fn" -> "variants_vcfgzs",
Also need to update accordingly in the app script.

yihchii commented on sequencing/vcf_slice/dxapp.json:
```diff
@@ -1,69 +1,90 @@
     {
       "name": "vcfidx_fn",
-      "label": "VCF Index Files",
+      "label": "VCF index files",
```
"vcfidx_fn" -> "variants_vcfgztbis",
If input is a \*.vcf.gz file, then the index filename should be \*.vcf.tbi.  (Likely the case)
If input is a \*.vcf file, then the index filename should be \*.vcf.idx.

Also need to update variables accordingly in the app script.

msimbirsky replied:
A common filename for *.vcf.gz file is *.vcf.gz.tbi

yihchii commented on sequencing/vcf_slice/dxapp.json:
```diff
@@ -1,69 +1,90 @@
   ],
   "outputSpec": [
     {
       "name": "vcf_out",
```
"vcf_out" -> "out_variants_vcfgzs"
The variable name seems to be misplaced with "out_variants_vcfgztbis" (below)

Also need to update variables accordingly in the app script.

yihchii commented on sequencing/vcf_slice/dxapp.json:
```diff
@@ -1,69 +1,90 @@
+      "patterns": ["*.vcf.gz.tbi"]
     },
     {
       "name": "vcfidx_out",
```
"vcfidx_out" -> "out_variants_vcfgztbis"
The variable name seems to be misplaced with "out_variants_vcfgzs" (above)

Also need to update variables accordingly in the app script.

yihchii commented on sequencing/vcf_slice/dxapp.json:
```diff
@@ -1,69 +1,90 @@
+        "*": { "instanceType": "azure:mem2_ssd1_x4" }
+      },
+      "assetDepends": [{"id": "record-FGpfZ1j95qzXV43bP5y5JKgV"}]
+    }
```
Please update record IDs to:
aws:us-east-1: record-FJ3zf8j08zVZFzqYG88Pbk7z
azure:westus: record-FJ4058j9Vf9ZBBVB2Q7b3VQz

yihchii commented on sequencing/vcf_slice/dxapp.json:
```diff
@@ -1,69 +1,90 @@
-  "authorizedUsers": ["org-geisinger_affiliate", "org-geisinger", "org-support", "user-codushlaine", "user-semanti.mukherjee", "user-alexander.li", "user-shannon.bruse", "user-jan.freudenberg", "user-crisvh"]
+  "details": {
+        "upstreamVersion": "",
+        "whatsNew": "",
```
"whatsNew": "* 1.0.0 First version of the Slice VCF File(s) app."

yihchii reviewed (commented):
Some comments:

1. Please update the app version to 1.0.0.
2. Please follow the variable nomenclature.
3. There seems to have two output variables naming mixed. Please confirm and fix.
4. Please update to the latest asset record IDs.

slzarate commented on sequencing/vcf_slice/Readme.md:
```diff
@@ -1,21 +1,32 @@
 # Slice VCF File(s) (DNAnexus Platform App)
 
-Return a small section of a VCF file (similar to tabix).  For large output, many small regions, or subsetting samples, use subset_vcf instead.
+Return a small section of a VCF file (similar to tabix).  For large output, 
```
Make into 1 line

msimbirsky replied:
Markdown ignores linebreaks, so this change wouldn't do anything.

slzarate commented on sequencing/vcf_slice/Readme.md:
```diff
@@ -1,21 +1,32 @@
-<!-- /dx-header -->
+## App Inputs
 
+This app requires 1 or more variant files in VCF format to be filtered to the
```
Make into 1 line, also specify that it's (`*.vcf.gz`)

slzarate commented on sequencing/vcf_slice/Readme.md:
```diff
@@ -1,21 +1,32 @@
+## App Inputs
 
+This app requires 1 or more variant files in VCF format to be filtered to the
+target region and a corresponding Tabix index file as well as a region file
```
Specify that it's (`*.vcf.gz.tbi`) and whatever the format for the region file is

slzarate commented on sequencing/vcf_slice/Readme.md:
```diff
@@ -1,21 +1,32 @@
+
+## App Output
+
+The app returns VCF files in `*.vcf.gz` format, with corresponding tabix index files.
```
Specify (`*.vcf.gz.tbi`)

Commit:
Add code review changes

- update readme
- update version number
- update whatsNew info
- update input names and filetypes

msimbirsky commented:
@slzarate @yihhwang Please see changes in response to your comments
